src/yaml-language-support/yaml-support.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+
+import { parse, findNodeAtPosition, util as yamlUtil } from 'node-yaml-parser';
+
+export interface YamlCache {
+    // the document model list represents the yaml text
+    yamlDocs: any[];
+
+    // lineLens contains the converted line length of each lines, it is used for converting from
+    // vscode position to the inner position in yaml element model.
+    lineLens: number[];
+
+    // the version of the document to avoid duplicate work on the same text
+    version: number;
+}
+
+export interface YamlMatchedElement {
+    // the document model list represents the yaml text
+    readonly yamlDocs: any[];
```

Comment:
I don't understand how an array of documents fits into a "matched element" interface.

Reply:
I will remove the yamlDocs from YamlMatchedElement